pkg/alertmanager/multitenant.go

```diff
@@ -1089,6 +1186,39 @@ func (am *MultitenantAlertmanager) getPerUserDirectories() map[string]string {
 	return result
 }
 
+// UpdateState implements the Alertmanager service.
+func (am *MultitenantAlertmanager) ReadState(ctx context.Context, req *alertmanagerpb.ReadStateRequest) (*alertmanagerpb.ReadStateResponse, error) {
+	userID, err := tenant.TenantID(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("no user id found in context")
```

Comment:
We should return the actual error:

```suggestion
		return nil, err
```

Reply:
Same fix for `UpdateState`: https://github.com/cortexproject/cortex/pull/3972